tests/specs/lemmas.k

```diff
@@ -330,6 +341,44 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ] [ N3 := BUF3:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ] [ N3 := BUF3           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], N, K ) => #range( #range( M, N, K ) [ 0 := BUF ], 0, K ) 
+      requires N >Int 0 [simplification]
+
+    rule #range ( M, N, K ) => #range( #drop(N, M), 0, K ) 
+      requires N >Int 0 [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], 0, K ) => M [ N := BUF ]
+      requires (#sizeByteArray( M ) ==Int K andBool #sizeByteArray( BUF ) <=Int K)
+        orBool (#sizeByteArray( M ) <=Int K andBool #sizeByteArray( BUF ) ==Int K)  [simplification]
+
+    rule M [ 0 := BUF:ByteArray ] => #padRightToWidth( #sizeByteArray( M ), BUF )
+      requires M ==K #padToWidth( #sizeByteArray( M ), .ByteArray ) 
+      andBool  #sizeByteArray( BUF ) <=Int #sizeByteArray( M )  [simplification]
+
+    rule #asWord( #padToWidth ( _, BUF ) ) => #asWord( BUF ) [simplification]
+
+    rule #range ( padRightBytes ( BUF:ByteArray, K, 0 ), 0, K ) => #padRightToWidth ( K, BUF ) [simplification]
+
+    rule #range (#padToWidth(K, BUF:ByteArray), 0, K) => #padToWidth(K, BUF)  [simplification]
+
+    rule #range ( b"", _, _) => b"" [simplification]
+
```

Comment:
Counterexample: `#range(b"", 0, 1)` should produce `b"\x00"`.

Reply:
Add a test for this please.

Reply:
By this do you just mean a claim like `claim <k> runLemma(#range( b"", 0, 1)) => doneLemma(b"\x00") ... </k>`? I am a bit confused about how to think about tests here. Are we generally supposed to just write the tests based on what we would expect a term to evaluate to by inspecting the semantics, and then let the lemma apply to it?

Reply:
Yes, those small proof claims are tests of our simplification lemmas, and your test looks good. It can go in `tests/specs/functional/lemmas-spec.k`.